LinaTheExplorer/LinaTheExplorer/2. Hi bye device/RulesForDevice.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Hi_bye_device
+{
+    class RulesForDevice
+    {
+        public static List<Car> GenerateRules(List<Car> listOfCars)
+        {
+            var listOfCarsArray = listOfCars.ToArray();
+
+            var rulesForSelecting = new Car[listOfCarsArray.Length];
+
+            for (int i = 0; i < rulesForSelecting.Length; i++)
+            {
+                rulesForSelecting[i] = listOfCarsArray[i];
+
+
+                var tooTalkative = rulesForSelecting[i].WelcomingMessage.Length > 20;
+                var ashole = rulesForSelecting[i].WelcomingMessage.Contains('*');
+                var vowelsInName = CheckVowels(rulesForSelecting[i].DriversName.ToString()) <= 2;
+                var sameFirstAndLastNumber = rulesForSelecting[i].CarNumber.ToString().EndsWith(rulesForSelecting[i].CarNumber.ToString().ElementAt(0));
+                var sumOfElementsInNumber = SumNumber(rulesForSelecting[i].CarNumber) > 20 && SumNumber(rulesForSelecting[i].CarNumber) <30;
+
+                var notPrimary = !ashole ^ !tooTalkative ^ vowelsInName ^ !sameFirstAndLastNumber ^ !sumOfElementsInNumber;
```

Comment:
this line made me think hard :)
It may contain a bug, please review if you combine the rules correctly